plinth/modules/dynamicdns/dynamicdns.py

```diff
@@ -0,0 +1,394 @@
+                                             required=False)
+
+    dynamicdns_domain = TrimmedCharField(
+        label=_('Domain Name'),
+        help_text=_(hlp_domain),
+        required=False,
+        validators=[
+            validators.RegexValidator(r'^[\w-]{1,63}(\.[\w-]{1,63})*$',
+                                      _('Invalid domain name'))])
+
+    dynamicdns_user = TrimmedCharField(
+        label=_('Username'),
+        required=False,
+        help_text=_(hlp_user))
+
+    dynamicdns_secret = TrimmedCharField(
+        label=_('Password'), widget=forms.PasswordInput(),
+        required=False,
+        help_text=_(hlp_secret))
+
+    showpw = forms.BooleanField(label=_('show password'),
+                                required=False)
+
+    dynamicdns_ipurl = TrimmedCharField(
+        label=_('IP check URL'),
+        required=False,
+        help_text=_(hlp_ipurl),
+        validators=[
+            validators.URLValidator(schemes=['http', 'https', 'ftp'])])
+
+    def clean(self):
+        cleaned_data = super(ConfigureForm, self).clean()
+        dynamicdns_secret = cleaned_data.get('dynamicdns_secret')
+        dynamicdns_update_url = cleaned_data.get('dynamicdns_update_url')
+        dynamicdns_user = cleaned_data.get('dynamicdns_user')
+        dynamicdns_domain = cleaned_data.get('dynamicdns_domain')
+        dynamicdns_server = cleaned_data.get('dynamicdns_server')
+        service_type = cleaned_data.get('service_type')
+        old_dynamicdns_secret = self.initial['dynamicdns_secret']
+
+        """clear the fields which are not in use"""
+        if service_type == '1':
+            dynamicdns_update_url = ""
+        else:
+            dynamicdns_server = ""
+
+        if cleaned_data.get('enabled'):
+            """check if gnudip server or update URL is filled"""
+            if not dynamicdns_update_url and not dynamicdns_server:
+                raise forms.ValidationError('please give update URL or \
+                                             a GnuDIP Server')
+                LOGGER.info('no server address given')
+
+            if dynamicdns_server and not dynamicdns_user:
+                raise forms.ValidationError('please give GnuDIP username')
+
+            if dynamicdns_server and not dynamicdns_domain:
+                raise forms.ValidationError('please give GnuDIP domain')
+
+            """check if a password was set before or a password is set now"""
+            if (dynamicdns_server and not dynamicdns_secret
+               and not old_dynamicdns_secret):
+                raise forms.ValidationError('please give a password')
+                LOGGER.info('no password given')
+
+
+@login_required
+@package.required('ez-ipupdate')
+def configure(request):
+    """Serve the configuration form"""
+    status = get_status()
+    form = None
+
+    if request.method == 'POST':
+        form = ConfigureForm(request.POST, initial=status, prefix='dynamicdns')
```

Comment:
There is only one form on this page so you could leave out the prefix, that would make statements like https://github.com/freedombox/Plinth/pull/81/files#diff-7d03a225ca7030e5d0ebe5b7123156d4R86 shorter (the `dynamicdns-dynamicdns` part)